src/index.ts

```diff
@@ -28,147 +28,91 @@ export interface INetPrinter {
+    const { promiseOrTimeout, timeoutId } = promiseWithTimeout(
+      new Promise((resolve, reject) =>
+        RNBLEPrinter.connectAndSend(
+          bdAddress,
+          data.toString('base64'),
+          (printer: INetPrinter) => resolve(printer),
+          (error: Error) => reject(error)
+        )
       )
-    ),
-
-  closeConn: (): Promise<void> =>
-    new Promise((resolve) => {
-      RNBLEPrinter.closeConn();
-      resolve();
-    }),
-
-  print: (text: string): void => {
-    if (Platform.OS === 'ios') {
-      RNBLEPrinter.printRawData(text, (error: Error) => console.warn(error));
-    } else {
-      RNBLEPrinter.printRawData(text, (error: Error) => console.warn(error));
-    }
+    );
+    return new Promise((resolve, reject) =>
+      promiseOrTimeout
+        .then((printer: INetPrinter) => resolve(printer))
+        .catch((error: Error) => reject(error))
+        .finally(() => clearTimeout(timeoutId))
+    );
   },
 };
 
 export const NetPrinter = {
-  init: (): Promise<void> =>
-    new Promise((resolve, reject) =>
-      RNNetPrinter.init(
-        () => resolve(),
-        (error: Error) => reject(error)
-      )
-    ),
-
-  getDeviceList: (): Promise<INetPrinter[]> =>
-    new Promise((resolve, reject) =>
-      RNNetPrinter.getDeviceList(
-        (printers: INetPrinter[]) => resolve(printers),
-        (error: Error) => reject(error)
-      )
-    ),
-
-  connectPrinter: (host: string, port: number): Promise<INetPrinter> =>
-    new Promise((resolve, reject) =>
-      RNNetPrinter.connectPrinter(
-        host,
-        port,
-        (printer: INetPrinter) => resolve(printer),
-        (error: Error) => reject(error)
-      )
-    ),
-
-  closeConn: (): Promise<void> =>
-    new Promise((resolve) => {
-      RNNetPrinter.closeConn();
-      resolve();
-    }),
-
-  print: (data: Buffer): void => {
-    if (Platform.OS === 'ios') {
-      RNNetPrinter.printRawData(data.toString('hex'), (error: Error) =>
-        console.warn(error)
-      );
-    } else {
-      RNNetPrinter.printRawData(data.toString('base64'), (error: Error) =>
-        console.warn(error)
-      );
-      // Or may be we try to send byte data
-      // RNNetPrinter.printByteData(Object.values(text), (error: Error) =>
-      //    console.warn(error)
-      // );
-    }
-  },
-
   connectAndSend: (
     host: string,
     port: number,
     data: Buffer
   ): Promise<INetPrinter> => {
-    return new Promise((resolve, reject) =>
-      RNNetPrinter.connectAndSend(
-        host,
-        port,
-        Platform.OS === 'ios' ? data.toString('hex') : data.toString('base64'),
-        (printer: INetPrinter) => resolve(printer),
-        (error: Error) => reject(error)
+    const { promiseOrTimeout, timeoutId } = promiseWithTimeout(
+      new Promise((resolve, reject) =>
+        RNNetPrinter.connectAndSend(
+          host,
+          port,
+          data.toString('base64'),
```

Comment:
for iOS we need to send `hex` in the old world.

Will base64 work?

Reply:
Handled it in the next pr. (where both sdks are included)